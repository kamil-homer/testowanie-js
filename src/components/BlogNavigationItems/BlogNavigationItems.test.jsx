import { describe, expect, test } from "vitest";
import {
  BlogNavigationItems,
  MENU_ITEMS,
  MENU_TYPE,
} from "./BlogNavigationItems";
import { render, screen } from "@testing-library/react";

describe("BlogNavigationItems", () => {
  test("should render desktop navigation items as buttons", () => {
    const buttonLabel = MENU_ITEMS[0].label;

    render(<BlogNavigationItems menuType={MENU_TYPE.DESKTOP} />);

    const menuItem = screen.getByRole("button", { name: buttonLabel });

    expect(menuItem).toBeInTheDocument();
  });

  test("should render mobile navigation items as menuitems", () => {
    const buttonLabel = MENU_ITEMS[0].label;

    render(<BlogNavigationItems menuType={MENU_TYPE.MOBILE} />);

    const menuItem = screen.getByRole("menuitem", { name: buttonLabel });

    expect(menuItem).toBeInTheDocument();
  });
});

describe("BlogNavigationItems snapshots", () => {
  test("should render correct strucutre for desktop", () => {
    const { asFragment } = render(
      <BlogNavigationItems menuType={MENU_TYPE.DESKTOP} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
