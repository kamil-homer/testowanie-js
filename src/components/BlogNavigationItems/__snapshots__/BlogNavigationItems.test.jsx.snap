// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogNavigationItems snapshots > should render correct strucutre for desktop 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":r6:"
    tabindex="0"
    type="button"
  >
    Features
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":r7:"
    tabindex="0"
    type="button"
  >
    Testimonials
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":r8:"
    tabindex="0"
    type="button"
  >
    Highlights
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":r9:"
    tabindex="0"
    type="button"
  >
    Pricing
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":ra:"
    tabindex="0"
    type="button"
  >
    FAQ
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInfo css-1cn90u7-MuiButtonBase-root-MuiButton-root"
    id=":rb:"
    tabindex="0"
    type="button"
  >
    Blog
  </button>
</DocumentFragment>
`;
